cmake/host-tools.cmake

```diff
@@ -23,11 +23,12 @@ else()
     )
 
   if(WEST_PYTHON)
+    get_filename_component(WEST_PYTHON ${WEST_PYTHON} ABSOLUTE)
     if(NOT (${WEST_PYTHON} STREQUAL ${PYTHON_EXECUTABLE}))
```

Comment:
Are you sure that PYTHON_EXECUTABLE matches the style of the above get_filename_component in all configurations, or do we need something like this?

```
get_filename_component(PYTHON_EXECUTABLE_ABSOLUTE "${PYTHON_EXECUTABLE}" ABSOLUTE)
if (NOT ${WEST_PYTHON} STREQUAL ${PYTHON_EXECUTABLE_ABSOLUTE})
```

I am not sure either way, just asking to confirm.

Reply:
Would it be worth separating this into an extension function,  `zephyr_same_paths()`, or so, by the way?

Reply:
> Are you sure that PYTHON_EXECUTABLE matches the style of the above get_filename_component in all configurations, or do we need something like this?

yes, because this looks up the the filename in the same manner as `find_program` used by FindPython.
This is different to `file(TO_CMAKE_PATH ..)` which simply converts the slashes.

Reply:
> yes, because this looks up the the filename in the same manner as `find_program` used by FindPython.

Does that allow the user to override the search by manually specifying PYTHON_EXECUTABLE at the command line, which makes the input format an arbitrary path rather than a well-formed CMake path? I seem to recall some find package libraries having features like that but I don't know how this one works.

Reply:
> Does that allow the user to override the search by manually specifying PYTHON_EXECUTABLE at the command line,

Users should not override using `PYTHON_EXECUTABLE` but should use `PYTHON_PREFER_EXECUTABLE`.
`PYTHON_EXECUTABLE` is the old variable used by Zephyr.
`find_package(Python3)` sets `Python3_EXECUTABLE` which is then used to `PYTHON_EXECUTABLE`, used in Zephyr.

The value of `PYTHON_PREFER_EXECUTABLE` is correctly adjusted to proper path separators, as it is fed into `find_program` here: https://github.com/zephyrproject-rtos/zephyr/blob/master/cmake/python.cmake#L17-L41

Reply:
> Would it be worth separating this into an extension function, `zephyr_same_paths()`, or so, by the way?

not much to gain, cause you would still need to have a dedicated return variable, and then test like:
```
zephyr_same_path(RESULT ${PATH1} ${PATH2})
if(${RESULT})
  # Do something
end()
```
cause CMake doesn't allow:
```
if(zephyr_same_path(${PATH1} ${PATH2}))
  # Do something
endif()
```

Reply:
> The value of `PYTHON_PREFER` is correctly adjusted to proper path separators, as it is fed into `find_program` here: https://github.com/zephyrproject-rtos/zephyr/blob/master/cmake/python.cmake#L17-L41

Perfect, thanks for explaining.